decider-bcl2fastq/src/main/java/ca/on/oicr/pde/deciders/utils/BarcodeAndBasesMask.java

```diff
@@ -0,0 +1,178 @@
+package ca.on.oicr.pde.deciders.utils;
+
+import ca.on.oicr.pde.deciders.data.Barcode;
+import ca.on.oicr.pde.deciders.data.BasesMask;
+import ca.on.oicr.pde.deciders.exceptions.DataMismatchException;
+import com.google.common.base.Joiner;
+import com.google.common.collect.Iterables;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ *
+ * @author mlaszloffy
+ */
+public class BarcodeAndBasesMask {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add a private constructor to hide the implicit public one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1118)